### ISSUE TITLE
Classic blocks show as Custom HTML in the Site Editor

### ISSUE THREAD

**Issue by carolinan** [WordPress/gutenberg#50358]
### Description

When you go through the Navigation Sidebar, select Navigation, and then select a link to a page, that page is opened in a view where you can see the template and edit the page content at the same time.

I can expand the list view and select and edit the blocks that are inside the post content block.
But, instead of classic blocks, the HTML block is used. This is confusing.

I think the classic block needs to be enabled in the Site Editor again.

### Step-by-step reproduction instructions

1. Activate a block theme.
2. Setup any navigation block.
3. Create a page and add a classic block.
4. Go to Appearance > Editor > Navigation. Here, select your page from the list. Select Edit.
5. Notice that there is a Custom HTML block nested inside the post content.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by Mamaduka** [comment]
The Site Editor doesn't support Classic block. It uses HTML as a fallback - #48129.

**Comment by carolinan** [comment]
Yes, but users should not need to be aware of that? It should just work, editing different blocks in your content depending on what screen you are viewing is confusing.

**Comment by draganescu** [comment]
This is a horrible experience when migrating from a classic theme isn't it? I don't think we should just hand wave at the technical limitation. I don't think including the classic block is the right way, maybe just use an uneditable preview of sorts?

**Comment by carolinan** [comment]
I wonder if it should only show a "Convert to blocks" button... but I guess that has already been tested, discussed, and discarded.

**Comment by carolinan** [comment]
**_If_** editing the classic block content as HTML risks breaking formatting, plugins, etc then I think making it uneditable would be better.

**Comment by talldan** [comment]
I think the reason behind this is that the classic block isn't used in the site editor. The classic block isn't used in the site editor because of the extra baggage of the tinymce dependency (and in the past, the iframe compatibility) AFAIK.

I think this is a bigger issue when the site editor is displaying post/page content, as there's a chance it has been migrated from a classic editor.

For templates, it's a bit different, maybe the html isn't from the classic editor. The classic editor might not be the best way to represent that content.

Maybe there is a better option than the html block, not sure. I think the issue is lacking proposals. HTML block is probably a little better now in that it shows the preview by default and not the code.
